build/Targets/Packages.props

```diff
@@ -326,13 +326,31 @@
     <PrivateVisualStudioPackage Include="VSLangProj140"/>
     <PrivateVisualStudioPackage Include="VsWebsite.Interop"/>
   </ItemGroup>
-  
+
   <PropertyGroup>
     <UsingToolPdbConverter>true</UsingToolPdbConverter>
     <UsingToolSymbolUploader>true</UsingToolSymbolUploader>
     <UsingToolNuGetRepack>true</UsingToolNuGetRepack>
   </PropertyGroup>
 
+  <!--
+    Non-default certificates.
+  -->
+  <ItemGroup>
+    <FileSignInfo Include="Microsoft.CodeAnalysis.ExpressionEvaluator.ExpressionCompiler.dll" CertificateName="WindowsPhone623"/>
+    <FileSignInfo Include="Microsoft.CodeAnalysis.ExpressionEvaluator.FunctionResolver.dll" CertificateName="WindowsPhone623"/>
+    <FileSignInfo Include="Microsoft.CodeAnalysis.CSharp.ExpressionEvaluator.ExpressionCompiler.dll" CertificateName="WindowsPhone623"/>
+    <FileSignInfo Include="Microsoft.CodeAnalysis.VisualBasic.ExpressionEvaluator.ExpressionCompiler.dll" CertificateName="WindowsPhone623"/>
+
+    <FileSignInfo Include="Microsoft.CodeAnalysis.ExpressionEvaluator.ResultProvider.dll" PublicKeyToken="31bf3856ad364e35" TargetFramework=".NETStandard,Version=v1.3" CertificateName="WindowsPhone623"/>
+    <FileSignInfo Include="Microsoft.CodeAnalysis.CSharp.ExpressionEvaluator.ResultProvider.dll" PublicKeyToken="31bf3856ad364e35" TargetFramework=".NETStandard,Version=v1.3" CertificateName="WindowsPhone623"/>
+    <FileSignInfo Include="Microsoft.CodeAnalysis.VisualBasic.ExpressionEvaluator.ResultProvider.dll" PublicKeyToken="31bf3856ad364e35" TargetFramework=".NETStandard,Version=v1.3" CertificateName="WindowsPhone623"/>
+
+    <FileSignInfo Include="Microsoft.CodeAnalysis.ExpressionEvaluator.ResultProvider.dll" PublicKeyToken="31bf3856ad364e35" TargetFramework=".NETFramework,Version=v2.0" CertificateName="MicrosoftSHA1Win8WinBlue"/>
+    <FileSignInfo Include="Microsoft.CodeAnalysis.CSharp.ExpressionEvaluator.ResultProvider.dll" PublicKeyToken="31bf3856ad364e35" TargetFramework=".NETFramework,Version=v2.0" CertificateName="MicrosoftSHA1Win8WinBlue"/>
+    <FileSignInfo Include="Microsoft.CodeAnalysis.VisualBasic.ExpressionEvaluator.ResultProvider.dll" PublicKeyToken="31bf3856ad364e35" TargetFramework=".NETFramework,Version=v2.0" CertificateName="MicrosoftSHA1Win8WinBlue"/>
+  </ItemGroup>
```

Comment:
Why do we specify these here and not in the project files of these components?

Reply:
Signing is not run in the context of project files. It's in the "outer build", so that it can aggregate all build outputs.

Reply:
We could potentially invent an attribute `AuthenticodeCertificate` that we would emit in the assemblies instead of this configuration. Then the sign tool could read that attribute.

Reply:
I'm not following you here. Do we have a write up of how the signing functions now? It's confiusung that we'd have to special case these 

---
In reply to: [215462442](https://github.com/dotnet/roslyn/pull/29622#discussion_r215462442) [](ancestors = 215462442)

Reply:
Not sure if it's worth it though, it should be pretty rare to use non-default certificates.

Reply:
That seems like a better approach. Having signing be specified globally here for just a few values feels like the design is off. 

---
In reply to: [215462686](https://github.com/dotnet/roslyn/pull/29622#discussion_r215462686) [](ancestors = 215462686)

Reply:
If we have to specify this globally but it's only used in the signing phase why not put this in a separate file? It feels odd to have the certificate information for an EE DLL be information available in the build when I'm compiling csc.exe. 

---
In reply to: [215462780](https://github.com/dotnet/roslyn/pull/29622#discussion_r215462780) [](ancestors = 215462780)

Reply:
Well, all the -Version properties are also available globally to projects that do not use them.

Reply:
But yes, we could create another file that gets imported only to the Sign.proj.

Reply:
The -Version properties are global because there is a reasonable chance they could be used by any project. There is no reason for these to be used in any project except Sign.proj

---
In reply to: [215464673](https://github.com/dotnet/roslyn/pull/29622#discussion_r215464673) [](ancestors = 215464673)

Reply:
The target framework is used to determine the cert, so binaries of the same name and PKT that target different framework can have different cert. That's actually the case with the EE assemblies.